src/python/pants/backend/python/util_rules/local_dists_pep660.py

```diff
@@ -0,0 +1,539 @@
+    Digest,
+    DigestSubset,
+    FileContent,
+    MergeDigests,
+    PathGlobs,
+    RemovePrefix,
+    Snapshot,
+)
+from pants.engine.process import Process, ProcessResult
+from pants.engine.rules import Get, MultiGet, collect_rules, rule
+from pants.engine.target import AllTargets, Target, Targets, WrappedTarget, WrappedTargetRequest
+from pants.engine.unions import UnionMembership
+from pants.util.docutil import doc_url
+from pants.util.frozendict import FrozenDict
+from pants.util.logging import LogLevel
+from pants.util.osutil import is_macos_big_sur
+from pants.util.strutil import softwrap
+
+logger = logging.getLogger(__name__)
+
+
+# We just use DistBuildRequest directly instead of adding a PEP660BuildRequest
+
+
+@dataclass(frozen=True)
+class PEP660BuildResult:  # based on DistBuildResult
+    output: Digest
+    # Relpaths in the output digest.
+    editable_wheel_path: str | None
+
+
+# PEP 660 defines `prepare_metadata_for_build_editable`. If PEP 660 is not
+# supported, we fall back to PEP 517's `prepare_metadata_for_build_wheel`.
+# PEP 517, however, says that method is optional. So finally we fall back
+# to using `build_wheel` and then extract the dist-info directory and then
+# delete the extra wheel file. Most people shouldn't hit that path (I hope).
+# NOTE: PEP 660 does not address the `.data` directory, so we ignore it.
+_BACKEND_WRAPPER_BOILERPLATE = """
+# DO NOT EDIT THIS FILE -- AUTOGENERATED BY PANTS
+
+import base64
+import hashlib
+import os
+import shutil
+import zipfile
+import {build_backend_module}
+
+backend = {build_backend_object}
+
+dist_dir = "{dist_dir}"
+build_dir = "build"
+pth_file_path = "{pth_file_path}"
+wheel_config_settings = {wheel_config_settings_str}
+tags = "{tags}"
+direct_url = "{direct_url}"
+
+# Python 2.7 doesn't have the exist_ok arg on os.makedirs().
+for d in (dist_dir, build_dir):
+    try:
+        os.makedirs(d)
+    except OSError as e:
+        if e.errno != errno.EEXIST:
+            raise
+
+prepare_metadata = getattr(
+    backend,
+    "prepare_metadata_for_build_editable",  # PEP 660
+    getattr(backend, "prepare_metadata_for_build_wheel", None),  # PEP 517
+)
+if prepare_metadata is not None:
+    print("prepare_metadata: " + str(prepare_metadata))
+    metadata_path = prepare_metadata(build_dir, wheel_config_settings)
+else:
+    # Optional PEP 517 method not defined. Use build_wheel instead.
+    wheel_path = backend.build_wheel(build_dir, wheel_config_settings)
+    full_wheel_path = os.path.join(build_dir, wheel_path)
+
+    with zipfile.ZipFile(full_wheel_path, "r") as whl:
+        dist_info_files = [n for n in whl.namelist() if ".dist-info/" in n]
+        whl.extractall(build_dir, dist_info_files)
+        metadata_path = os.path.dirname(dist_info_files[0])
+
+pkg_version = metadata_path.replace(".dist-info", "")
+if "-" in pkg_version:
+    pkg, version = pkg_version.split("-")
+else:
+    # an old backend that doesn't conform to the latest specs
```

Comment:
```suggestion
    # An old backend that doesn't conform to the latest specs.
```

I'm a stickler for grammar... But complete sentences do help convey to the reader that this is a complete thought.